### PR TITLE
Add tests for backends contract

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Test that updates are no-op for non-existing databases or collections.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — DISMISSED** [review]
👌

**Review by chilagrow — DISMISSED** [review]
Thank you for this :pray:

File: internal/backends/collection_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+				otherCollName := collName + "_other"
+
+				db, err := b.Database(dbName)
+				require.NoError(t, err)
+
+				// to create database
+				err = db.CreateCollection(ctx, &backends.CreateCollectionParams{
+					Name: otherCollName,
+				})
+				require.NoError(t, err)
```
Comment (chilagrow) [inline internal/backends/collection_test.go:93]: Oddly this test is failing https://github.com/FerretDB/FerretDB/actions/runs/6341505616/job/17225292812?pr=3456

Do you need to drop the database before or after the test?

**Review by noisersup — COMMENTED** [review]
Looks good to me, apart from failing test

**Pull request merged**